tests/test_serialization.py

```diff
@@ -99,3 +100,259 @@ def test_serializing_polyfield_by_parent_type(field):
     rect_dict = field.serialize('shape', marshmallow_sticker)
 
     assert rect_dict == {"length": 4, "width": 10, "color": "blue"}
+
+
+def test_serializing_with_modification():
+    import marshmallow
+
+    def create_label_schema(schema):
+        class LabelSchema(marshmallow.Schema):
+            type = marshmallow.fields.String()
+            value = schema()
+
+        return LabelSchema
+
+    class_to_schema = {
+        text_type: marshmallow.fields.String,
+        int: marshmallow.fields.Integer,
+    }
+
+    name_to_class = {
+        u'str': text_type,
+        u'int': int,
+    }
+
+    class_to_name = {
+        cls: name
+        for name, cls in name_to_class.items()
+    }
+    class_to_name[text_type] = u'str'
+
+    def serialization_schema(base_object, parent_obj):
+        cls = type(base_object)
+        schema = class_to_schema[cls]
+
+        label_schema = create_label_schema(schema=schema)
+
+        return label_schema()
+
+    def serialization_value(base_object, parent_obj):
+        cls = type(base_object)
+        name = class_to_name[cls]
+
+        return {'type': name, 'value': base_object}
+
+    def deserialization_schema(object_dict, parent_object_dict):
+        name = object_dict['type']
+        cls = name_to_class[name]
+        schema = class_to_schema[cls]
+
+        return schema()
+
+    def deserialization_value(object_dict, parent_object_dict):
+        return object_dict['value']
+
+    class TopClass:
+        def __init__(self, polyfield):
+            self.polyfield = polyfield
+
+        def __eq__(self, other):
+            if type(self) != type(other):
+                return False
+
+            return self.polyfield == other.polyfield
+
+    class TopSchema(marshmallow.Schema):
+        polyfield = PolyField(
+            serialization_schema_selector=serialization_schema,
+            deserialization_schema_selector=deserialization_schema,
+            serialization_value_modifier=serialization_value,
+            deserialization_value_modifier=deserialization_value,
+        )
+
+        @marshmallow.decorators.post_load
+        def make_object(self, data, many=None, partial=None):
+            return TopClass(**data)
+
+    top_schema = TopSchema()
+
+    top_class_str_example = TopClass(polyfield=u'abc')
+    top_class_str_example_dumped = top_schema.dump(top_class_str_example)
+    print(top_class_str_example_dumped)
```

Comment:
rather than printing I think you should validate the json that comes out with an assertion

Reply:
This is just exploratory code and I think it's fully replaced at this point by 'real' tests.  It's in the WIP list to be removed.